### ISSUE TITLE
Make Groups Useful: Item-level access controls

### ISSUE THREAD

**Issue by danhorst** [projecthydra-labs/hyku#612]
Issue #421 defined the requirements for managing groups throughout the Administrative UI. It did not address how groups would integrate with item-level access controls.

Group membership facilitates two kinds of actions:
1. Interacting with the repository system itself (another issue)
2. Interacting with content stored in the repository system (this issue)

If groups can be assigned to Works, Collections, and Administrative Sets then:
- The access control management UI for each of these types of content needs to include managing the association between groups and repository content needs to bubble up onto the appropriate pages.
- Because associating a group with private or restricted repository content may change what search results are displayed to a user Solr must be made aware of groups.

**Comment by mjgiarlo** [comment]
Blocked by #608.

**Comment by mjgiarlo** [comment]
No longer blocked. We think we may have the code and possibly the UI for this already. Requires testing/verification.
